### ISSUE TITLE
Panic when building distributive-rpm

### ISSUE THREAD

**Issue by langston-barrett** [asteris-llc/hammer#20]
I got the following panic on a fresh clone of distributive-rpm, with a clone distributive in the same directory. This happens with master and with @sehqlr's branch. 

```
distributive-rpm ➤ ./hammer/hammer-bin build              
INFO[0000] loading packages                              root=/home/siddharthist/Dropbox/code/asteris/distributive-rpm
INFO[0000] finished setup stage                          name=distributive stage=setup
INFO[0000] finished setup stage                          name=distributive stage=setup
ERRO[0000] build script exited and could not read stdout  error=open /home/siddharthist/Dropbox/code/asteris/distributive-rpm/logs/distributive-stderr-2015-12-28T16:30:14-06:00.log: no such file or directory name=distributive
panic: read |0: bad file descriptor

goroutine 49 [running]:
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc8202fbce0, 0x90e150, 0x6, 0x7fb33bd08328, 0xc820332038)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:79 +0x8f
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:314 +0x5aa

goroutine 1 [select]:
github.com/asteris-llc/hammer/hammer.(*Packager).Build(0xc8202de520, 0x7fb33bd0cff8, 0xc820011780, 0x4, 0x7fb33bd0cf01)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:87 +0x3e0
main.glob.func1(0xb60bc0, 0xb8c7c8, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/asteris/distributive-rpm/hammer/build.go:85 +0xa0b
github.com/spf13/cobra.(*Command).execute(0xb60bc0, 0xb8c7c8, 0x0, 0x0, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/spf13/cobra/command.go:534 +0x869
github.com/spf13/cobra.(*Command).Execute(0xb60d80, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/spf13/cobra/command.go:617 +0x46a
main.main()
        /home/siddharthist/Dropbox/code/asteris/distributive-rpm/hammer/main.go:54 +0xce

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/lib/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [select, locked to thread]:
runtime.gopark(0x9dc580, 0xc82002c728, 0x90dc18, 0x6, 0x42f618, 0x2)
        /usr/lib/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc82002c728, 0x0, 0x18)
        /usr/lib/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc82002c728)
        /usr/lib/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
        /usr/lib/go/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 7 [select]:
main.glob.func1.1(0xc8202d9620, 0xc820299250, 0x7fb33bd0cff8, 0xc820011780)
        /home/siddharthist/Dropbox/code/asteris/distributive-rpm/hammer/build.go:74 +0x1d6
created by main.glob.func1
        /home/siddharthist/Dropbox/code/asteris/distributive-rpm/hammer/build.go:83 +0x9c0

goroutine 8 [syscall]:
syscall.Syscall6(0x3d, 0x50f5, 0xc820331754, 0x0, 0xc820354990, 0x0, 0x0, 0xc820331728, 0xc820331730, 0xc820331720)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.wait4(0x50f5, 0xc820331754, 0x0, 0xc820354990, 0x90, 0x0, 0x0)
        /usr/lib/go/src/syscall/zsyscall_linux_amd64.go:172 +0x72
syscall.Wait4(0x50f5, 0xc82033179c, 0x0, 0xc820354990, 0xc820332098, 0x0, 0x0)
        /usr/lib/go/src/syscall/syscall_linux.go:256 +0x55
os.(*Process).wait(0xc8203069e0, 0x27, 0x0, 0x0)
        /usr/lib/go/src/os/exec_unix.go:22 +0x105
os.(*Process).Wait(0xc8203069e0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/doc.go:45 +0x2d
os/exec.(*Cmd).Wait(0xc82035c3c0, 0x0, 0x0)
        /usr/lib/go/src/os/exec/exec.go:380 +0x211
os/exec.(*Cmd).Run(0xc82035c3c0, 0x0, 0x0)
        /usr/lib/go/src/os/exec/exec.go:258 +0x64
os/exec.(*Cmd).CombinedOutput(0xc82035c3c0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/exec/exec.go:424 +0x310
github.com/asteris-llc/hammer/hammer.(*FPM).PackageFor(0xc8202ee240, 0xc820298b90, 0x3, 0x0, 0x0, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/fpm.go:65 +0x956
github.com/asteris-llc/hammer/hammer.(*Package).Package(0xc8200de600, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:361 +0x63
github.com/asteris-llc/hammer/hammer.(*Package).Package-fm(0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:146 +0x2e
github.com/asteris-llc/hammer/hammer.(*Package).BuildAndPackage(0xc8200de600, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:152 +0x3e7
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc8202de520, 0xc8202de5a0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:38 +0xf1
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1cf

goroutine 9 [select]:
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc8202de520, 0xc8202de5a0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:36 +0x2ac
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1cf

goroutine 10 [select]:
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc8202de520, 0xc8202de5a0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:36 +0x2ac
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1cf

goroutine 11 [syscall]:
syscall.Syscall6(0x3d, 0x50e4, 0xc8200eb904, 0x0, 0xc820366000, 0x0, 0x0, 0x4510a0, 0x7fb33bea44b8, 0xa)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.wait4(0x50e4, 0xc8200eb904, 0x0, 0xc820366000, 0x90, 0x0, 0x0)
        /usr/lib/go/src/syscall/zsyscall_linux_amd64.go:172 +0x72
syscall.Wait4(0x50e4, 0xc8200eb94c, 0x0, 0xc820366000, 0xc82002e7c8, 0x0, 0x0)
        /usr/lib/go/src/syscall/syscall_linux.go:256 +0x55
os.(*Process).wait(0xc8202deb60, 0x27, 0x0, 0x0)
        /usr/lib/go/src/os/exec_unix.go:22 +0x105
os.(*Process).Wait(0xc8202deb60, 0x0, 0x0, 0x0)
        /usr/lib/go/src/os/doc.go:45 +0x2d
os/exec.(*Cmd).Wait(0xc820016a00, 0x0, 0x0)
        /usr/lib/go/src/os/exec/exec.go:380 +0x211
github.com/asteris-llc/hammer/hammer.(*Package).Build(0xc8200de200, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:329 +0xa0b
github.com/asteris-llc/hammer/hammer.(*Package).Build-fm(0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:145 +0x2e
github.com/asteris-llc/hammer/hammer.(*Package).BuildAndPackage(0xc8200de200, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:152 +0x3e7
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc8202de520, 0xc8202de5a0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:38 +0xf1
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1cf

goroutine 24 [syscall]:
syscall.Syscall(0x0, 0x3, 0xc82035a000, 0x1000, 0x70, 0x7c2380, 0x1)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x3, 0xc82035a000, 0x1000, 0x1000, 0x7fb3386dba10, 0x0, 0x0)
        /usr/lib/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x3, 0xc82035a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/lib/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc82002e7b0, 0xc82035a000, 0x1000, 0x1000, 0xc820358000, 0x0, 0x0)
        /usr/lib/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc82002e7b0, 0xc82035a000, 0x1000, 0x1000, 0x859e20, 0x0, 0x0)
        /usr/lib/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc820342eb8, 0x1000)
        /usr/lib/go/src/bufio/scan.go:180 +0x877
github.com/asteris-llc/hammer/hammer.(*FileConsumer).HandleStream(0xc8203126c0, 0x90e150, 0x6, 0x7fb33bd08328, 0xc82002e7b0, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:62 +0x18f
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc8203126c0, 0x90e150, 0x6, 0x7fb33bd08328, 0xc82002e7b0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:77 +0x4d
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:314 +0x5aa

goroutine 25 [syscall]:
syscall.Syscall(0x0, 0x5, 0xc820368000, 0x1000, 0x70, 0x7c2380, 0x1)
        /usr/lib/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x5, 0xc820368000, 0x1000, 0x1000, 0x7fb3386db000, 0x0, 0x0)
        /usr/lib/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x5, 0xc820368000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
        /usr/lib/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc82002e7c0, 0xc820368000, 0x1000, 0x1000, 0xc820312c30, 0x0, 0x0)
        /usr/lib/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc82002e7c0, 0xc820368000, 0x1000, 0x1000, 0x859e20, 0x0, 0x0)
        /usr/lib/go/src/os/file.go:95 +0x8a
bufio.(*Scanner).Scan(0xc82003feb8, 0x1000)
        /usr/lib/go/src/bufio/scan.go:180 +0x877
github.com/asteris-llc/hammer/hammer.(*FileConsumer).HandleStream(0xc8203126c0, 0x90e140, 0x6, 0x7fb33bd08328, 0xc82002e7c0, 0x0, 0x0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:62 +0x18f
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc8203126c0, 0x90e140, 0x6, 0x7fb33bd08328, 0xc82002e7c0)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:77 +0x4d
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:321 +0x836

goroutine 50 [runnable]:
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc8202fbce0, 0x90e140, 0x6, 0x7fb33bd08328, 0xc820332048)
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/logs.go:76
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
        /home/siddharthist/Dropbox/code/go/src/github.com/asteris-llc/hammer/hammer/package.go:321 +0x836

goroutine 51 [runnable]:
os/exec.(*Cmd).Start.func1(0xc82035c3c0, 0xc8203067c0)
        /usr/lib/go/src/os/exec/exec.go:339
created by os/exec.(*Cmd).Start
        /usr/lib/go/src/os/exec/exec.go:341 +0x96d
```

**Comment by siddharthist** [comment]
When using "zsh" as the shell, I get a different error:
```
distributive-rpm ➤ hammer build --search="." --shell="zsh"
INFO[0000] loading packages                              root=.
INFO[0000] finished setup stage                          name=distributive stage=setup
INFO[0000] finished setup stage                          name=distributive stage=setup
ERRO[0000] build script exited with a non-zero exit code  error=<nil> name=distributive stderr=godep: No Godeps found (or in any parent directory)
 stdout=
INFO[0000] finished build stage                          name=distributive stage=build
ERRO[0000] build script exited with a non-zero exit code  error=<nil> name=distributive stderr=/tmp/hammer-distributive-script276803591/build:cd:1: no such file or directory: distributive/packaging//..
godep: No Godeps found (or in any parent directory)
 stdout=
INFO[0000] finished build stage                          name=distributive stage=build
ERRO[0000] failed to package                             error=exit status 1 name=distributive out={:timestamp=>"2015-12-28T16:33:16.246493-0600", :message=>"Invalid package configuration: Cannot package the path '/tmp/hammer-distributive-build795034440/distributive', does it exist?", :level=>:error}
 outType=rpm
ERRO[0000] could not complete stage                      error=exit status 1 name=distributive stage=package
ERRO[0000] failed to package                             error=exit status 1 name=distributive out={:timestamp=>"2015-12-28T16:33:16.263665-0600", :message=>"Invalid package configuration: Cannot package the path '/tmp/hammer-distributive-build760585468/distributive', does it exist?", :level=>:error}
 outType=rpm
ERRO[0000] could not complete stage                      error=exit status 1 name=distributive stage=package
```

**Comment by siddharthist** [comment]
And using @sehqlr's branch, I get this:
```
distributive-rpm ➤ go run hammer/*.go build --shell="zsh"            
INFO[0000] loading packages                              root=/home/siddharthist/Dropbox/code/asteris/distributive-rpm
INFO[0000] finished setup stage                          name=distributive stage=setup
INFO[0000] finished setup stage                          name=distributive stage=setup
ERRO[0000] build script exited with a non-zero exit code  error=<nil> name=distributive stderr=godep: No Godeps found (or in any parent directory)
 stdout=
INFO[0000] finished build stage                          name=distributive stage=build
ERRO[0000] failed to package                             error=exit status 1 name=distributive out={:timestamp=>"2015-12-28T16:34:15.171290-0600", :message=>"Invalid package configuration: Cannot package the path '/tmp/hammer-distributive-build089426285/distributive', does it exist?", :level=>:error}
 outType=rpm
ERRO[0000] could not complete stage                      error=exit status 1 name=distributive stage=package
INFO[0003] finished build stage                          name=distributive stage=build
ERRO[0004] failed to package                             error=exit status 1 name=distributive out={:timestamp=>"2015-12-28T16:34:18.505821-0600", :message=>"no value for epoch is set, defaulting to nil", :level=>:warn}
{:timestamp=>"2015-12-28T16:34:18.513128-0600", :message=>"no value for epoch is set, defaulting to nil", :level=>:warn}
{:timestamp=>"2015-12-28T16:34:18.513605-0600", :message=>"Need executable 'rpmbuild' to convert dir to rpm", :level=>:error}
 outType=rpm
ERRO[0004] could not complete stage                      error=exit status 1 name=distributive stage=package
exit status 1
```

**Comment by langston-barrett** [comment]
I get a similar error when building the `mesos` branch of mantl-packaging:

```
time="2016-01-26T03:09:54-05:00" level=info msg="loading packages" root="/vagrant/mesos" 
time="2016-01-26T03:09:54-05:00" level=info msg="finished setup stage" name=mesos-agent stage=setup 
time="2016-01-26T03:09:54-05:00" level=warning msg="build script not found. skipping build stage." name=mesos-agent 
time="2016-01-26T03:09:54-05:00" level=info msg="finished build stage" name=mesos-agent stage=build 
time="2016-01-26T03:09:54-05:00" level=info msg="finished setup stage" name=mesos stage=setup 
time="2016-01-26T03:09:55-05:00" level=info msg="finished package stage" name=mesos-agent stage=package 
time="2016-01-26T03:09:55-05:00" level=info msg="finished setup stage" name=mesos-agent-dynamic stage=setup 
time="2016-01-26T03:09:55-05:00" level=warning msg="build script not found. skipping build stage." name=mesos-agent-dynamic 
time="2016-01-26T03:09:55-05:00" level=info msg="finished build stage" name=mesos-agent-dynamic stage=build 
time="2016-01-26T03:09:56-05:00" level=info msg="finished package stage" name=mesos-agent-dynamic stage=package 
time="2016-01-26T03:09:56-05:00" level=info msg="finished setup stage" name=mesos-master stage=setup 
time="2016-01-26T03:09:56-05:00" level=warning msg="build script not found. skipping build stage." name=mesos-master 
time="2016-01-26T03:09:56-05:00" level=info msg="finished build stage" name=mesos-master stage=build 
time="2016-01-26T03:09:56-05:00" level=info msg="finished package stage" name=mesos-master stage=package 
time="2016-01-26T03:09:56-05:00" level=info msg="finished setup stage" name=mesos-master-dynamic stage=setup 
time="2016-01-26T03:09:56-05:00" level=warning msg="build script not found. skipping build stage." name=mesos-master-dynamic 
time="2016-01-26T03:09:56-05:00" level=info msg="finished build stage" name=mesos-master-dynamic stage=build 
time="2016-01-26T03:09:57-05:00" level=info msg="finished package stage" name=mesos-master-dynamic stage=package 
time="2016-01-26T03:09:57-05:00" level=info msg="finished setup stage" name=marathon stage=setup 
panic: read |0: bad file descriptor

goroutine 148 [running]:
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc209f48000, 0x8b9910, 0x6, 0x7f47c2647d10, 0xc2080383a0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/logs.go:79 +0x90
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:322 +0x7f3

goroutine 1 [select, 9 minutes]:
github.com/asteris-llc/hammer/hammer.(*Packager).Build(0xc20801f1c0, 0x7f47c26485c0, 0xc208070580, 0x2, 0xc20802bd01)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/packager.go:87 +0x3fc
main.func·001(0xaf2aa0, 0xc20801e940, 0x0, 0x2)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/build.go:85 +0xb75
github.com/spf13/cobra.(*Command).execute(0xaf2aa0, 0xc20801e7c0, 0x2, 0x2, 0x0, 0x0)
    /home/vagrant/go/src/github.com/spf13/cobra/command.go:569 +0x82f
github.com/spf13/cobra.(*Command).ExecuteC(0xaf2ca0, 0xaf2aa0, 0x0, 0x0)
    /home/vagrant/go/src/github.com/spf13/cobra/command.go:656 +0x504
github.com/spf13/cobra.(*Command).Execute(0xaf2ca0, 0x0, 0x0)
    /home/vagrant/go/src/github.com/spf13/cobra/command.go:615 +0x3a
main.main()
    /home/vagrant/go/src/github.com/asteris-llc/hammer/main.go:57 +0xe6

goroutine 5 [syscall, 11 minutes]:
os/signal.loop()
    /usr/lib/golang/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
    /usr/lib/golang/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [select, 11 minutes]:
main.func·005()
    /home/vagrant/go/src/github.com/asteris-llc/hammer/build.go:74 +0x206
created by main.func·001
    /home/vagrant/go/src/github.com/asteris-llc/hammer/build.go:83 +0xb2d

goroutine 7 [syscall, 9 minutes]:
syscall.Syscall6(0x3d, 0x26fb, 0xc208075984, 0x0, 0xc2080c6510, 0x0, 0x0, 0x7f47c2636000, 0x4059a1, 0xc20807e7a0)
    /usr/lib/golang/src/syscall/asm_linux_amd64.s:46 +0x5
syscall.wait4(0x26fb, 0xc208075984, 0x0, 0xc2080c6510, 0x90, 0x0, 0x0)
    /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:124 +0x79
syscall.Wait4(0x26fb, 0xc2080759cc, 0x0, 0xc2080c6510, 0x0, 0x0, 0x0)
    /usr/lib/golang/src/syscall/syscall_linux.go:224 +0x60
os.(*Process).wait(0xc20801f080, 0x0, 0x0, 0x0)
    /usr/lib/golang/src/os/exec_unix.go:22 +0x103
os.(*Process).Wait(0xc20801f080, 0xc20801eda0, 0x0, 0x0)
    /usr/lib/golang/src/os/doc.go:45 +0x3a
os/exec.(*Cmd).Wait(0xc20800c780, 0x0, 0x0)
    /usr/lib/golang/src/os/exec/exec.go:364 +0x23c
github.com/asteris-llc/hammer/hammer.(*Package).Build(0xc208032d80, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:330 +0x9bc
github.com/asteris-llc/hammer/hammer.*Package.Build·fm(0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:146 +0x39
github.com/asteris-llc/hammer/hammer.(*Package).BuildAndPackage(0xc208032d80, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:153 +0x41f
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc20801f1c0, 0xc20801f260)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/packager.go:38 +0xf0
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1e7

goroutine 8 [runnable]:
bytes.(*Buffer).Write(0xc209ca00e0, 0xc209d72000, 0x7e83b, 0x80000, 0x1, 0x1, 0xc2080d2000)
    /usr/lib/golang/src/bytes/buffer.go:125
fmt.Fprintf(0x7f47c2648058, 0xc209ca00e0, 0x89c6d0, 0x2, 0xc2080ed798, 0x1, 0x1, 0x1edc2, 0x0, 0x0)
    /usr/lib/golang/src/fmt/print.go:189 +0xac
github.com/Sirupsen/logrus.(*TextFormatter).appendKeyValue(0xc20801ea40, 0xc209ca00e0, 0x8b9950, 0x6, 0x76bae0, 0xc2080d2050)
    /home/vagrant/go/src/github.com/Sirupsen/logrus/text_formatter.go:147 +0x27d
github.com/Sirupsen/logrus.(*TextFormatter).Format(0xc20801ea40, 0xc209caa140, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/vagrant/go/src/github.com/Sirupsen/logrus/text_formatter.go:91 +0x6df
github.com/Sirupsen/logrus.(*Entry).Reader(0xc209caa140, 0xc209f3a102, 0x0, 0x0)
    /home/vagrant/go/src/github.com/Sirupsen/logrus/entry.go:44 +0x70
github.com/Sirupsen/logrus.Entry.log(0xc20800a200, 0xc209f3a150, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc2080d2002, 0xc209f3a180, ...)
    /home/vagrant/go/src/github.com/Sirupsen/logrus/entry.go:94 +0x286
github.com/Sirupsen/logrus.(*Entry).Error(0xc209caa100, 0xc2080edc68, 0x1, 0x1)
    /home/vagrant/go/src/github.com/Sirupsen/logrus/entry.go:145 +0x93
github.com/asteris-llc/hammer/hammer.(*Package).Build(0xc208033d40, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:346 +0x1129
github.com/asteris-llc/hammer/hammer.*Package.Build·fm(0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:146 +0x39
github.com/asteris-llc/hammer/hammer.(*Package).BuildAndPackage(0xc208033d40, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:153 +0x41f
github.com/asteris-llc/hammer/hammer.(*Packager).startWorker(0xc20801f1c0, 0xc20801f260)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/packager.go:38 +0xf0
created by github.com/asteris-llc/hammer/hammer.(*Packager).Build
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/packager.go:79 +0x1e7

goroutine 54 [syscall]:
syscall.Syscall(0x0, 0x3, 0xc2080ab74b, 0x8b5, 0x74b, 0x1000, 0x0)
    /usr/lib/golang/src/syscall/asm_linux_amd64.s:21 +0x5
syscall.read(0x3, 0xc2080ab74b, 0x8b5, 0x8b5, 0x74b, 0x0, 0x0)
    /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:867 +0x6e
syscall.Read(0x3, 0xc2080ab74b, 0x8b5, 0x8b5, 0x701, 0x0, 0x0)
    /usr/lib/golang/src/syscall/syscall_unix.go:136 +0x58
os.(*File).read(0xc208038100, 0xc2080ab74b, 0x8b5, 0x8b5, 0xc20807b9e0, 0x0, 0x0)
    /usr/lib/golang/src/os/file_unix.go:191 +0x5e
os.(*File).Read(0xc208038100, 0xc2080ab74b, 0x8b5, 0x8b5, 0xc2080d2020, 0x0, 0x0)
    /usr/lib/golang/src/os/file.go:95 +0x91
bufio.(*Scanner).Scan(0xc2080a8f00, 0xc2080ab000)
    /usr/lib/golang/src/bufio/scan.go:180 +0x688
github.com/asteris-llc/hammer/hammer.(*FileConsumer).HandleStream(0xc2080d4cf0, 0x8b9950, 0x6, 0x7f47c2647d10, 0xc208038100, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/logs.go:62 +0x225
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc2080d4cf0, 0x8b9950, 0x6, 0x7f47c2647d10, 0xc208038100)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/logs.go:77 +0x50
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:315 +0x580

goroutine 55 [syscall, 2 minutes]:
syscall.Syscall(0x0, 0x6, 0xc20800f000, 0x1000, 0x1a, 0x80f, 0x0)
    /usr/lib/golang/src/syscall/asm_linux_amd64.s:21 +0x5
syscall.read(0x6, 0xc20800f000, 0x1000, 0x1000, 0x1a, 0x0, 0x0)
    /usr/lib/golang/src/syscall/zsyscall_linux_amd64.go:867 +0x6e
syscall.Read(0x6, 0xc20800f000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
    /usr/lib/golang/src/syscall/syscall_unix.go:136 +0x58
os.(*File).read(0xc208038110, 0xc20800f000, 0x1000, 0x1000, 0xc208087d40, 0x0, 0x0)
    /usr/lib/golang/src/os/file_unix.go:191 +0x5e
os.(*File).Read(0xc208038110, 0xc20800f000, 0x1000, 0x1000, 0xc2080dfc70, 0x0, 0x0)
    /usr/lib/golang/src/os/file.go:95 +0x91
bufio.(*Scanner).Scan(0xc2080a8f80, 0xc20800f7f1)
    /usr/lib/golang/src/bufio/scan.go:180 +0x688
github.com/asteris-llc/hammer/hammer.(*FileConsumer).HandleStream(0xc2080d4cf0, 0x8b9910, 0x6, 0x7f47c2647d10, 0xc208038110, 0x0, 0x0)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/logs.go:62 +0x225
github.com/asteris-llc/hammer/hammer.(*FileConsumer).MustHandleStream(0xc2080d4cf0, 0x8b9910, 0x6, 0x7f47c2647d10, 0xc208038110)
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/logs.go:77 +0x50
created by github.com/asteris-llc/hammer/hammer.(*Package).Build
    /home/vagrant/go/src/github.com/asteris-llc/hammer/hammer/package.go:322 +0x7f3
```

**Comment by sehqlr** [comment]
What does it look like when you test it against my PR? I removed the call to the `MustHandleStream` method from that line in my version.

**Comment by ChrisAubuchon** [comment]
Maybe your build script is failing. The code path that the `build script exited and could not read stdout` log entry logs from only occurs when `cmd.Wait()` errors.

**Comment by langston-barrett** [comment]
@sehqlr As you say, it doesn't fail when your branch in #19 is used. @ChrisAubuchon That may be true, it's hard to tell where the failure occurs in the build script if the logging panics.

**Comment by BrianHicks** [comment]
FYI, this is the only place in hammer that panics can happen: https://github.com/asteris-llc/hammer/blob/master/hammer/logs.go#L79

I need to redo how logging is handled, which will fix this (hopefully)

**Comment by langston-barrett** [comment]
Should be closed by #22.
